shell/platform/android/io/flutter/embedding/engine/FlutterEngine.java

```diff
@@ -159,6 +162,32 @@ public FlutterEngine(
         automaticallyRegisterPlugins);
   }
 
+  /**
+   * Same as {@link #FlutterEngine(Context, String[], boolean)} with added support for configuring
+   * whether the engine will receive restoration data.
+   *
+   * <p>When the engine is configured to receive restoration data {@code
+   * RestorationChannel.setRestorationData(byte[] data)} must be called to provide the restoration
+   * data. All requests to get restoration data from the framework will be blocked until that method
+   * is called. If the engine is configured to wait for restoration data, but it turns out later
+   * that no restoration data has been provided by the operating system, that method must still be
+   * called with null as an argument to indicate "no data".
```

Comment:
I think this could still use some more details. e.g. the user might think at this point, what's the purpose for setting this constructor flag? If I had to provide restoration data, why can't I just call setRestorationData as needed rather than trying to forward declare this ahead of time?